distribution/pull_v2_windows.go

```diff
@@ -23,15 +23,20 @@ func (ld *v2LayerDescriptor) Descriptor() distribution.Descriptor {
 }
 
 func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekCloser, error) {
+	blobs := ld.repo.Blobs(ctx)
+	rsc, err := blobs.Open(ctx, ld.digest)
+
 	if len(ld.src.URLs) == 0 {
-		blobs := ld.repo.Blobs(ctx)
-		return blobs.Open(ctx, ld.digest)
+		return rsc, err
 	}
 
-	var (
-		err error
-		rsc distribution.ReadSeekCloser
-	)
+	// We're done if the registry has this layer.
+	if err == nil {
+		if _, err = rsc.Seek(0, os.SEEK_SET); err == nil {
```

Comment:
Can we do `if rsc != nil || len(ld.src.URLs) ==0 { return rsc, err }` instead?
This is a bit confusing at least for me.